lib/ansible/playbook/role_include.py

```diff
@@ -43,6 +43,11 @@ class IncludeRole(TaskInclude):
     circumstances related to the `- include_role: ...`
     """
 
+    BASE = set(['name', 'role'])
+    FROM_ARGS = set(['tasks_from', 'vars_from', 'defaults_from'])
+    OTHER_ARGS = set(['private', 'allow_duplicates'])
+    VALID_ARGS = frozenset(BASE.union(FROM_ARGS.union(OTHER_ARGS)))
```

Comment:
Maybe a comment for each of BASE, FROM_ARGS, OTHER_ARGS to say where they're coming from (I assume that's the reason to keep them separate)